private/bufpkg/bufplugin/bufplugindocker/buildkit.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufplugindocker
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/moby/buildkit/session"
+	"go.uber.org/zap"
+)
+
+const (
+	pathBuildkitNodeID = ".buildkit_node_id"
+	// Matches length of node ID in Docker CLI
+	nodeIDLength = 32
+)
+
+func createSession(ctx context.Context, logger *zap.Logger, contextDir, configDirPath string) (*session.Session, error) {
+	sharedKey := getBuildSharedKey(logger, contextDir, configDirPath)
+	s, err := session.NewSession(ctx, filepath.Base(contextDir), sharedKey)
+	if err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+func getBuildSharedKey(logger *zap.Logger, dir, configDirPath string) string {
```

Comment:
nit: We usually explicitly type the type of the parameter in every position, regardless if the same types are adjacent. Also, we should name parameters consistently across the functions that interact with it.
```suggestion
func getBuildSharedKey(logger *zap.Logger, contextDir string, configDirPath string) string {
```